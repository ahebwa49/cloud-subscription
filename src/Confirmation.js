import React, { useState } from "react";
import "./Confirmation.scss";

const Confirmation = () => {
  const [email, setEmail] = useState("");

  const onEmailChange = (e) => {
    console.log(e.target.value);
    setEmail(e.target.value);
  };

  return (
    <div className="Confirmation">
      <h3>Confirmation</h3>
      <div className="ConfirmationPrice">Price per GB:</div>
      <div className="ConfirmationTotalPrice">Total price:</div>
      <form className="ConfirmationForm">
        <div className="ConfirmationFormGroup">
          <label htmlFor="email" className="ConfirmationFormGroup__label">
            Email:
          </label>
          <input
            type="email"
            name="email"
            placeholder="Enter your email..."
            required
            value={email}
            onChange={onEmailChange}
          />
        </div>
        <button>Submit</button>
      </form>
    </div>
  );
};
export default Confirmation;
