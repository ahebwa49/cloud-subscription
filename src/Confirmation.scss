.Confirmation {
  display: flex;
  flex-direction: column;
  margin-top: 200px;
  padding: 1rem;
  width: 450px;
  height: 200px;
  border-radius: 4px;
  border: 1px solid #dadada;
}
.ConfirmationFormGroup {
  margin-bottom: 1rem;
}
.ConfirmationPrice {
  margin-bottom: 1rem;
}
.ConfirmationTotalPrice {
  margin-bottom: 1rem;
}
