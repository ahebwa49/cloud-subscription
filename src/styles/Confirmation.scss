.Confirmation {
  display: flex;
  flex-direction: column;
  margin-top: 150px;
  padding: 1rem;
  padding-top: 0;
  width: 450px;
  height: 330px;
  border-radius: 4px;
  border: 1px solid #dadada;
}
.ConfirmationFormGroup {
  margin-bottom: 1rem;
}
.ConfirmationSummary {
  margin-bottom: 1rem;
}
.ConfirmationError {
  color: red;
}
