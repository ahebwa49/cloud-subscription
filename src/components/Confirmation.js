import React from "react";
import "../styles/Confirmation.scss";
import { calculateTotal } from "../utils/index";

const Confirmation = ({
  onEmailChange,
  onTermsAndConditionsChange,
  handleFormSubmit,
  email,
  termsAndConditions,
  selectedPlan,
  upfrontPayment,
  storage,
}) => {
  const handleSubmit = (e) => {
    e.preventDefault();
    handleFormSubmit();
  };

  let totalPrice = calculateTotal(
    storage,
    selectedPlan.price_usd_per_gb,
    upfrontPayment
  );

  return (
    <div className="Confirmation">
      <h4>Confirmation</h4>
      <div className="ConfirmationSummary">
        <span>
          Subscription Plan:{" "}
          <span className="ConfirmationSummaryDuration">
            {selectedPlan.duration_months}
          </span>
          months,
        </span>
        <span className="ConfirmationSummaryPrice">
          {selectedPlan.price_usd_per_gb}
        </span>
        USD per GB
        <br />
        <br />
        <span>
          Amount of gigabytes in a cloud:{" "}
          <span className="ConfirmationSummaryStorage">{storage}</span>GB
        </span>
        <br />
        <br />
        <span>
          Upfront Payment:{" "}
          <span className="ConfirmationSummaryUpfront">{upfrontPayment}</span>
        </span>
        <br />
        <br />
        <span>Total Price: {totalPrice}USD</span>
      </div>
      <form className="ConfirmationForm" onSubmit={handleSubmit}>
        <div className="ConfirmationFormGroup">
          <label htmlFor="email" className="ConfirmationFormGroup__label">
            Email:{" "}
          </label>
          <input
            type="email"
            name="email"
            placeholder="Enter your email..."
            required
            value={email}
            onChange={(e) => onEmailChange(e)}
          />
        </div>
        <div className="ConfirmationFormGroup">
          <input
            type="checkbox"
            name="termsAndConditions"
            required
            checked={termsAndConditions === true}
            value={termsAndConditions}
            onChange={(e) => onTermsAndConditionsChange(e)}
          />
          <label htmlFor="email" className="ConfirmationFormGroup__label">
            Terms and conditions agreement
          </label>
        </div>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};
export default Confirmation;
