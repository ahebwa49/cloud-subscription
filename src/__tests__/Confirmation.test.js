import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Confirmation from "../components/Confirmation";
import confirmationData from "./data/confirmationData";

describe("create App snapshot", () => {
  test("create snapshot", () => {
    let tree = renderer
      .create(
        <Confirmation
          onEmailChange={() => {}}
          onTermsAndConditionsChange={() => {}}
          handleFormSubmit={() => {}}
          email="ahebwa49@gmail.com"
          termsAndConditions={true}
          selectedPlan={{ duration_months: 12, price_usd_per_gb: 2 }}
          storage={50}
          upfrontPayment="yes"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

let tree;
for (let data of confirmationData) {
  let { selectedPlan, storage, upfrontPayment } = data;

  describe("Consistency in Payment component", () => {
    beforeEach(() => {
      tree = render(
        <Confirmation
          onCardNumberChange={() => {}}
          onExpirationDateChange={() => {}}
          onSecurityCodeChange={() => {}}
          cardNumber={1234}
          expirationDate={"2021-06-30"}
          securityCode={1234}
          selectedPlan={selectedPlan}
          storage={storage}
          upfrontPayment={upfrontPayment}
          error="All fields must be completed inorder to proceed to next step"
        />
      );
    });
    test("Ensure that the payment plan displayed in the UI is the right one", () => {
      let duration = document.querySelector(".ConfirmationSummaryDuration");
      let price = document.querySelector(".ConfirmationSummaryPrice");
      expect(parseInt(duration.innerHTML)).toBe(selectedPlan.duration_months);
      expect(parseInt(price.innerHTML)).toBe(selectedPlan.price_usd_per_gb);
    });
    test("Ensure that the storage displayed in the UI is the right one", () => {
      let elemStorage = document.querySelector(".ConfirmationSummaryStorage");
      expect(parseInt(elemStorage.innerHTML)).toBe(storage);
    });
    test("Ensure that the upfront payment displayed in the UI is the right one", () => {
      let elemUpfrontPayment = document.querySelector(
        ".ConfirmationSummaryUpfront"
      );
      expect(elemUpfrontPayment.innerHTML).toBe(upfrontPayment);
    });
  });
}
