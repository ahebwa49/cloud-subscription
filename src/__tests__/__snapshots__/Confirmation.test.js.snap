// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create App snapshot create snapshot 1`] = `
<div
  className="Confirmation"
>
  <h4>
    Confirmation
  </h4>
  <div
    className="ConfirmationSummary"
  >
    <span>
      Subscription Plan:
       
      <span
        className="ConfirmationSummaryDuration"
      >
        12
      </span>
      months,
    </span>
    <span
      className="ConfirmationSummaryPrice"
    >
      2
    </span>
    USD per GB
    <br />
    <br />
    <span>
      Amount of gigabytes in a cloud:
       
      <span
        className="ConfirmationSummaryStorage"
      >
        50
      </span>
      GB
    </span>
    <br />
    <br />
    <span>
      Upfront Payment:
       
      <span
        className="ConfirmationSummaryUpfront"
      >
        yes
      </span>
    </span>
    <br />
    <br />
    <span>
      Total Price: 
      100
      USD
    </span>
  </div>
  <form
    className="ConfirmationForm"
    onSubmit={[Function]}
  >
    <div
      className="ConfirmationFormGroup"
    >
      <label
        className="ConfirmationFormGroup__label"
        htmlFor="email"
      >
        Email:
         
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Enter your email..."
        required={true}
        type="email"
        value="ahebwa49@gmail.com"
      />
    </div>
    <div
      className="ConfirmationFormGroup"
    >
      <input
        checked={true}
        name="termsAndConditions"
        onChange={[Function]}
        required={true}
        type="checkbox"
        value={true}
      />
      <label
        className="ConfirmationFormGroup__label"
        htmlFor="email"
      >
        Terms and conditions agreement
      </label>
    </div>
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
